Mathlib/Analysis/Analytic/Binomial.lean

```diff
@@ -0,0 +1,386 @@
+/-
+Copyright (c) 2024 Vasilii Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasilii Nesterov
+-/
+import Mathlib.Analysis.ODE.Gronwall
+import Mathlib.Analysis.SpecialFunctions.Pow.Deriv
+import Mathlib.RingTheory.Binomial
+import Mathlib.Tactic.MoveAdd
+
+/-!
+
+# Binomial Series
+
+We introduce the binomial series:
+$$
+\sum_{k=0}^{\infty} \; \binom{a}{k} \; x^k = 1 + a x + \frac{a(a-1)}{2!} x^2 +
+  \frac{a(a-1)(a-2)}{3!} x^3 + \cdots
+$$
+where $a$ is an element of a normed field $\mathbb{K}$,
+and $x$ is an element of a normed algebra over $\mathbb{K}$.
+
+## Main Statements
+
+* `binomialSeries_radius_ge_one`: The radius of convergence of the binomial series is at least $1$.
+* `binomialSum_eq_rpow`: The series converges to $(1 + x)^a$ for real $a$ and $|x| < 1$.
+
+## Implementation Details
+
+We use `RCLike 𝕂` to leverage the fact that `‖(n : 𝕂)‖ = n` for natural numbers `n`. Without this,
+we cannot apply the `descPochhammer_bound_ascPochhammer` bound in our proof.
+
+## TODO
+
+* Generalize `binomialSeries_radius_ge_one` from `RCLike 𝕂` to `NontriviallyNormedField 𝕂`.
+* Prove that the radius of convergence of the binomial series is *exactly* $1$.
+* Prove the `binomialSum_eq_cpow` variant of `binomialSum_eq_rpow` for complex $a$ and $x$.
+
+-/
+
+open scoped Nat
+
+universe u v
+
+theorem Ring.choose_eq_div {𝕂 : Type v} [Field 𝕂] [CharZero 𝕂]
+    {a : 𝕂} {n : ℕ} :
+    Ring.choose a n = (n ! : 𝕂)⁻¹ • (descPochhammer ℤ n).smeval a := by
+  rw [Ring.descPochhammer_eq_factorial_smul_choose]
+  trans (n ! : 𝕂)⁻¹ • ((n ! : 𝕂) • Ring.choose a n)
+  · rw [smul_smul]
+    rw [inv_mul_cancel₀]
+    · simp
+    rw [Nat.cast_ne_zero]
+    exact Nat.factorial_ne_zero n
+  · congr
+    apply Nat.cast_smul_eq_nsmul
+
+theorem ascPochhammer_nonneg {R : Type u} [LinearOrderedCommRing R] {a : R} {n : ℕ} (ha : 0 ≤ a) :
+    0 ≤ (ascPochhammer ℕ n).smeval a := by
+  cases n with
+  | zero => simp
+  | succ m =>
+    simp [ascPochhammer_succ_right, Polynomial.smeval_mul, Polynomial.smeval_natCast]
+    apply mul_nonneg
+    · exact ascPochhammer_nonneg ha
+    · linarith
+
+section
+
+variable {𝕂 : Type v} [NormedField 𝕂]
+
+/-- Binomial series:
+$$
+\sum_{k=0}^{\infty} \; \binom{a}{k} \; x^k = 1 + a x + \frac{a(a-1)}{2!} x^2 +
+  \frac{a(a-1)(a-2)}{3!} x^3 + \cdots
+$$
+-/
+noncomputable def binomialSeries [CharZero 𝕂] (𝔸 : Type u) [NormedDivisionRing 𝔸] [Algebra 𝕂 𝔸]
```

Comment:
This looks very nice!  However, I would like to use this series outside the `NormedField` context.  Could we have something like:
```
noncomputable def PowerSeries.binomialSeries {R : Type v} [Ring R] [BinomialRing R] {A : Type u} [One A] [SMul R A] (r : R) :
    PowerSeries A :=
  mk fun n => (Ring.choose r n) • (1 : A)
```
and turn it into a `FormalMultilinearSeries` later?

Reply:
Thank you! I'm not sure about this. I'm not very familiar with the `PowerSeries` API. It seems there isn't an easy way to convert a `PowerSeries` to a `FormalMultilinearSeries` so far. This should probably be done in a separate PR.

Reply:
Okay, don't worry about it.